src/utils/in-memory-cache.ts

```diff
@@ -0,0 +1,65 @@
+import { idb } from "@/common/db";
+
+const PACMAN_WALLETS = "pacman-wallets";
+
+const inMemoryCache = () => {
+  const cache: Map<string, unknown> = new Map<string, unknown>();
+  return {
+    get: function (key: string): any {
+      return cache.get(key);
+    },
+    set: function (key: string, val: unknown) {
+      cache.set(key, val);
+    },
+  };
+};
+
+export const cache = inMemoryCache();
+
+export const updateWalletCache = async (address: string) => {
+  await saveWallet(address);
+  const walletExists: Record<string, boolean> = cache.get(PACMAN_WALLETS);
+  if (walletExists) {
+    walletExists[address] = true;
+    cache.set(PACMAN_WALLETS, walletExists);
+  } else {
+    cache.set(PACMAN_WALLETS, { [address]: true });
+  }
+};
+
+export const getCacheWallets = async (): Promise<Record<string, boolean>> => {
+  let wallets: string[] = cache.get(PACMAN_WALLETS);
+  if (!wallets) {
+    wallets = await idb.many("select address from pacman_wallets where status = 1");
+  }
+  return (
+    wallets.reduce((acc: Record<string, boolean>, address) => {
+      acc[address] = true;
+      return acc;
+    }, {}) ?? {}
+  );
+};
+
+export const saveWallet = async (address: string) => {
+  return idb.none(
+    `
+      INSERT INTO pacman_wallets (
```

Comment:
Can we avoid adding pacman related information on the indexer? @shubhampatwa

Reply:
@ptsayli @shubhampatwa I've pushed a commit to this branch changing name of the table to `tracked_wallets`. This commit also fixes a minor bug'